distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/WaitCondition.java

```diff
@@ -0,0 +1,122 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.fail;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import org.hamcrest.Matcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class used to wait for a certain condition to be {@code true} or a timeout to expire. The
+ * condition is checked at regular interval. Both the timeout and polling interval can be
+ * configured.
+ */
+public class WaitCondition {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WaitCondition.class);
+
+    private static final long DEFAULT_TIMEOUT = SECONDS.toMillis(5);
+
+    private static final long DEFAULT_POLLING_INTERVAL = MILLISECONDS.toMillis(500);
+
+    private String description;
+
+    private long timeoutMs;
+
+    private long pollingIntervalMs;
+
+    /**
+     * Creates a instance of this class using the default timeout (5s) and polling interval (500ms).
+     */
+    public WaitCondition(String description) {
+        this.description = description;
+        this.timeoutMs = DEFAULT_TIMEOUT;
+        this.pollingIntervalMs = DEFAULT_POLLING_INTERVAL;
+    }
+
+    /**
+     * Sets the timeout on this {@link WaitCondition}.
+     *
+     * @param timeout timeout to use
+     * @param unit    timeout unit
+     * @return the current {@link WaitCondition} object
+     */
+    public WaitCondition timeoutAfter(long timeout, TimeUnit unit) {
+        this.timeoutMs = unit.toMillis(timeout);
+        return this;
+    }
+
+    /**
+     * Sets the polling interval this {@link WaitCondition} will use to re-evaluate the wait
+     * condition.
+     *
+     * @param pollingInterval polling interval to use
+     * @param unit            polling interval unit
+     * @return the current {@link WaitCondition} object
+     */
+    public WaitCondition pollEvery(long pollingInterval, TimeUnit unit) {
+        this.pollingIntervalMs = unit.toMillis(pollingInterval);
+        return this;
+    }
+
+    /**
+     * Waits until the value returned by the {@link Callable} satisfies the {@link Matcher}'s
+     * condition, or the timeout expires. The wait condition will be re-evaluated every polling
+     * interval. If the timeout expires or an exception occurs while waiting, the current test
+     * will be failed.
+     *
+     * @param <T>                   type of the value returned by the {@link Callable}
+     * @param currentValueRetriever object to call to retrieve the value to wait on
+     * @param matchCondition        matcher used to determine whether the condition has been
+     *                              met or not
+     */
+    public <T> void waitFor(Callable<T> currentValueRetriever, Matcher<T> matchCondition) {
```

Comment:
We should be able to write as complex a `Predicate` as needed, but it would require us to actually write them instead of just using the hamcrest builtins. I guess it depends whether we prefer depending on hamcrest for this (assuming it might be extracted to its own library project).